### DISCUSSION TITLE
Report a Bug - Guideline

### DISCUSSION THREAD

**Discussion by satyamvats5** [0xTheProDev/Operational-Transformation#30] [Bug Report]
Report the bugs here and the maintainers will create an issue if it's a relevant one.

**NOTE:-** Don't create the issue directly.

Follow the below given format for reporting the Bugs.

**Describe the bug**

**To Reproduce**
Mention the steps required to reproduce the bug.

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):** 

**Smartphone (please complete the following information):** 

**Additional context**

**Comment by zaptrem** [comment]
Hello,

Apologies for posting this here, lmk if there's a better place to ask:

I'm trying to use OT's firebase-plaintext to implement a shared rich text editor using React Native's TextInput. However, after creating a FirebaseAdapter it seemed I also needed to implement an IEditorAdapter, which seems extremely complicated. Are there any resources I could read to better wrap my head around this? I assumed I would just need to plug in onChangeRecieved and onChangeCreated callbacks.

**Reply by 0xTheProDev** [reply]
Hi, apologies for inconvenience. OT package comes with out of box with Firebase Adapter to be plugged into Plaintext Editor. You can check here on what are implementation needed for editor adapter: https://github.com/0xTheProDev/Operational-Transformation/blob/main/packages/plaintext-editor/src/editor-adapter.ts

**Comment by anonymous9915** [comment]
Describe the bug
If the history is not empty, and then if I make any changes then it is throwing an error and also changes are not getting saved to the firebase.
Uncaught (in promise): Invalid Operation Encountered: The Operation recieved was either Invalid or Corrupted, please retry!
Invalid Operation Encountered: The Operation recieved was either Invalid or Corrupted, please retry!

The base length of the second operation has to be the target length of the first operation
Assertion Failed: The base length of the second operation has to be the target length of the first operation

Retain expects a positive integer.
Assertion Failed: Retain expects a positive integer.

To Reproduce
Create the firepad using FireMonacoEditor. then write something to the editor and refresh the page, you will get one of those errors.

Expected behavior
Data should be saved to the firebase without throwing errors.

Screenshots
![image](https://user-images.githubusercontent.com/87244355/173988111-5c2e0c4a-876f-420d-a8f8-da52376ee590.png)


Desktop (please complete the following information):

OS: Windows
Browser chrome
Version: 102.0.5005.115 (Official Build) (64-bit)
Framework: Angular 12

**Reply by 0xTheProDev** [reply]
Hi, OT supports data being present in Firebase earlier. In case if you have content in the editor, I would suggest you to copy out the data and set it later after OT has been initialised.
If above does not fix the problem, kindly try setting line endings to `\n`.

**Comment by asoran** [comment]
**Describe the bug**
I can't seem to use multiple instance of monaco with the firebase adapter (in a react project).
What I need is to have 2 components (Live1 and Live2) that loads and instantiate 2 monaco in the **same** page, connected to the same firebase project (but different path of course). Also both will be multi-tabbed (so multi-model).
When I have 1 editor loaded (either Live1 or Live2), it works perfectly fine. But when the 2 are loaded, non works.... It seems that the database adapter gets confused?? I made sure that they use 2 different model by doing `onMount={(editor, mon) => editor.setModel(mon.editor.createModel(...))}`.

**To Reproduce**
Create 2 components/files, in both files initialise the firebase adapter (`new FirebaseAdapter` from `@otjs/firebase-plaintext`), the editor adapter (`new MonacoAdapter` from `@otjs/monaco`) (I use the monaco editor react binding from `@monaco-editor/react`) and then the editor client (`new EditorClient` from `@otjs/plaintext-editor`). Configure firebase access tokens etc, load the 2 components/files in the dom: nothing works as expected (I can see that some stuff are created in firebase but then it breaks and the history gets frozen, no websocket request is sent anymore.

**Expected behavior**
I would like both instances to work correctly, and no I don't want them to be in different tabs, I need them both loaded at once (the same time), next to each other.

~**Screenshots**~

**Desktop**
*OS*: MacOs Monteney
*Browser*: Google Chrome
*Version*: 109.0.5414.87

~**Additional context**~

Thanks.

**Reply by 0xTheProDev** [reply]
Can you try with `@otjs/firebase-monaco` and instantiate with each editors separately. If that does not work, we can syncup.

**Reply by tyagirajat200** [reply]
Hi @0xTheProDev ,
I am also facing the same issue with @otjs/firebase-monaco.
https://github.com/0xTheProDev/Operational-Transformation/discussions/114#discussion-4437175

**Reply by asoran** [reply]
Hello @tyagirajat200 sorry for the long wait, I answered how I did it in your discussion!
